### PR TITLE
VarCreate bugfix

### LINKED ISSUE #1395: No VariableCreate event when dragging a block with a variable out of the flyout
### Problem statement

- [x] Bug report
- [ ] Feature request

Dragging a block with a variable out of the flyout fires a create event for the block but not for the variable, even though the variable is created on the workspace.

This is for a flyout block with a variable that is not already in the workspace.  e.g. 
image

### Expected Behavior

Variable creation events are fired when variables are created.

### Actual Behavior

Only a block creation event is fired.

### Steps to Reproduce

Enable "log events" on the playground.
Drag out a list block with a variable.

### Additional Information
```
Blockly.Flyout.prototype.createBlock = function(originalBlock) {
  var newBlock = null;
  Blockly.Events.disable();
  this.targetWorkspace_.setResizesEnabled(false);
  try {
    newBlock = this.placeNewBlock_(originalBlock);
    // Close the flyout.
    Blockly.hideChaff();
  } finally {
    Blockly.Events.enable();
  }

  if (Blockly.Events.isEnabled()) {
    Blockly.Events.setGroup(true);
    Blockly.Events.fire(new Blockly.Events.Create(newBlock));
  }
```

Blockly.Flyout.prototype.createBlock turns off events while placing the new block, then fires a single Create event that includes all of the blocks in the new stack.  It isn't aware of variables.  The variable create event is not fired from inside placeNewBlock because events are disabled.

I don't know why events are disabled here, instead of just grouped.

### PR DESCRIPTION
**Pull request by kchadha**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Resolves #1395 

### Proposed Changes

Fixes the bug where VarCreate does not get fired when dragging a block with a variable out into the workspace:

- check for any new variables when creating a new block from the flyout
- issue VarCreate event for any new variables (not previously in the workspace)

### Reason for Changes

Dragging out a block with a variable into the workspace was not issuing a VarCreate event even though a variable was getting added to the workspace.

### Test Coverage

Tried steps to reproduce issue #1395 and saw that the VarCreate event is getting fired once for each new variable being added to the workspace. VarCreate event does not retrigger when dragging blocks referring to existing variables out into the workspace.

Tested on:
* Desktop Chrome

### CODE REVIEW

File: core/flyout_base.js [hunk 1]
```diff
@@ -615,9 +616,28 @@ Blockly.Flyout.prototype.createBlock = function(originalBlock) {
     Blockly.Events.enable();
   }
 
+  var variablesAfterCreation = this.targetWorkspace_.getAllVariables();
```
Comment (rachel-fenichel) [inline core/flyout_base.js]: Make this into a helper function: `Blockly.Flyout.prototype.getAddedVariables(variablesBeforeCreation)` that returns the list of new variables.

**Comment by kchadha** [comment]
I signed it!

File: core/flyout_base.js (hunk 1 above)
Comment (kchadha) [reply core/flyout_base.js]: done.

File: core/flyout_base.js [hunk 2]
```diff
@@ -596,6 +596,33 @@ Blockly.Flyout.prototype.onMouseDown_ = function(e) {
   }
 };
 
+/**
+ * Helper function to get the list of variables that have been added to the
+ * workspace after adding a new block, using the given list of variables that
+ * were in the workspace before the new block was added.
+ * @param {!Array.<Blockly.VariableModel>} originalVariables The array of
```
Comment (rachel-fenichel) [inline core/flyout_base.js]: !Array.<!Blockly.VariableModel>, here and on the return.  And then we're done, and I'll merge it as soon as travis finishes!

**Pull request merged**